pylint/config.py

```diff
@@ -369,7 +370,7 @@ def __setitem__(self, path, config):
 @six.add_metaclass(abc.ABCMeta)
 class ConfigParser(object):
     def __init__(self):
-        self._option_definitions = {}
+        self._option_definitions = collections.OrderedDict()
```

Comment:
Changing this to OrderedDict to make tests more predictable. Should result in config files that are easier to diff too.